src/SIM/SIMbase.C

```diff
@@ -1625,34 +1625,60 @@ bool SIMbase::project (Matrix& ssol, const Vector& psol,
 }
 
 
-bool SIMbase::project (Vector& values, const RealFunc* f,
-                       int basis, int iField, int nFields, double time) const
+bool SIMbase::project (Vector& values, const FunctionBase* f,
+                       int basis, int iField, int nFields,
+                       SIMoptions::ProjectionMethod pMethod, double time) const
 {
   bool ok = true;
+  values.resize(f->dim()*mySam->getNoNodes());
```

Comment:
No, we can't use SAM::getNoDOFs() here, because the number of components per node in the secondary solutions that are projected, usually are different from the number of DOFs per node.
But I moved that resize to projectAnaSol instead to avoid the problem in Stokes/NavierStokes.